### ISSUE TITLE
Error about non-existing variable in keyword return value cannot be caught

### ISSUE THREAD

**Issue by serhatbolsu** [robotframework/robotframework#2217]
There is a case that I can not benefit from `Run Keyword and Ignore Error` and the execution immediately fails. 

For the below Example:

``` robotframework
    Set and Switch to Webview
      [Documentation]  Switch to Web view that is gathred from application and return its name
      ${all}  Get Contexts
      ${web}  Get Matches  ${all}  WEBVIEW*
      ${native}  Get Matches  ${all}  NATIVE*
      run keyword and ignore error  Switch To Context  ${web[0]}
      [Return]  ${web[0]}  ${native[0]}
```

`Switch to Context` is an library keyword from robotframework-appiumlibrary. However it could be any keyword. Here the execution does not gracefully fail when I put this keyword inside another keyword such as:

``` robotframework
    ${web_view}  ${native_view}  Wait Until Keyword Succeeds  30s  2s  Set and Switch to Webview
```

The  Set and Switch to Webview keyword fails with: 

```
FAIL    Replacing variables from keyword return value failed: Resolving variable '${web[0]}' failed: IndexError: list index out of range
```
- Test execution does not retry `Set and Switch to Webview` as it should with `Wait Until Keyword Succeeds`
- Same applies for `Run keyword and return status`

**Comment by stasiek** [comment]
You've said this will happen for any keyword which isn't true. Just take look at below testcase which will run as Passed, but internal keyword fails.

```robot
*** Test Cases ***
Test Run Keyword And Ignore Error
  Run Keyword And Ignore Error  Log  ${A[0]}

Test Run Keyword And Ignore Error In Keyword
  Test With Index

*** Keywords ***
Test With Index
  Run Keyword And Ignore Error  Log  ${A[0]}
```

**Comment by serhatbolsu** [comment]
@stasiek problem does not reveal itself in your case. Because test execution already finishes upon fail. 
Please try this:

```robotframework
*** Test Cases ***
Test Run Keyword And Ignore Error
  Run Keyword And Ignore Error  Log  ${A[0]}

Test Run Keyword And Ignore Error In Keyword
  Wait Until Keyword Succeeds  30s  2s  Test With Index

*** Keywords ***
Test With Index
  Run Keyword And Ignore Error  Log  ${A[0]}
  Sleep  5s
```

See if it actually does `Sleep` inside keyword and if the test executes for at least 30 seconds

**Comment by pekkaklarck** [comment]
This may have changed in RF 2.9.  Cannot verify right now.

**Comment by serhatbolsu** [comment]
@pekkaklarck I have just tested, `Sleep 1s` is executed however,  `Test With Index` keyword only run ones and test took 1.5 second

**Comment by pekkaklarck** [comment]
Why should it run more than once?

**Comment by serhatbolsu** [comment]
Well for logging it does not make sence, however in my example the response of `${all}  Get Contexts` changes and ones it receive the index, I would like it to pass.

There are other ways I can solve my problem which I already did, but I belive this is a problem.

**Comment by pekkaklarck** [comment]
Please clarify. What is the problem Wait Until Keyword Succeeds executing only once if the keyword it executes succeeds on the first round? Or do you still have some other problem?

**Comment by serhatbolsu** [comment]
Sorry I mislead you, the problem is about [Return] of keyword.

In such case:
```robotframework
Test Run Keyword And Ignore Error In Keyword
  run keyword and ignore error  Test With Index
  log  This is also executed

*** Keywords ***
Test With Index
  ${A}  create list
  Sleep  2s
  [Return]  ${A[0]}
```

2nd statement `log` can not execute. It seems that `run keyword and ignore error` can not handle return statement index failure

**Comment by pekkaklarck** [comment]
Ok. That happens also w/ 2.9? Could be that errors in return value are always considered syntax errors that cannot be silenced. If that's the case, I'm not sure how easy or hard it would be to fix. Need to take a closer look when I have my laptop in front of me.

**Comment by serhatbolsu** [comment]
@pekkaklarck 
I was working with 2.9.2 before, tested in 2.9 still happens.

**Comment by pekkaklarck** [comment]
I was able to reproduce the problem. Apparently variables in return values were not considered when fixing #1869.

**Comment by pekkaklarck** [comment]
We noticed same problem also with for loop values. Initially planned to just update this issue accordingly, but in the end decided to submit a new issue (#2256).
